File: challenges/010/cakes.py
def cakes(recipe, available):

    shared_keys = recipe.keys() & available.keys()

    return 0 if len(shared_keys) < len(recipe) else min(
        [available[key] // recipe[key] for key in shared_keys]
    )


# Simpler version:
def cakes(recipe, available):
    return min(available.get(key, 0) // amount for key, amount in recipe.items())

File: challenges/010/cakes.js
function cakes(recipe, available) {
    const sharedKeys = Object.keys(recipe).filter(key => Object.hasOwn(available, key));
    
    if (sharedKeys.length < Object.keys(recipe).length) {
      return 0;
    };

    return Math.min(...sharedKeys
      .map(key => Math.floor(available[key] / recipe[key]))
    );
}


// Simpler version:
function cakes(recipe, available) {
  return Math.min(
    ...Object.keys(recipe).map(key =>
      Math.floor((available[key] ?? 0) / recipe[key])
    )
  );
}